src/frontend/csrs-portal/src/app/auth/auth-config.module.ts

```diff
@@ -1,34 +1,48 @@
+/* eslint-disable arrow-body-style */
+import { HttpClient } from '@angular/common/http';
 import { NgModule } from '@angular/core';
-import { AuthModule, LogLevel } from 'angular-auth-oidc-client';
+import { AuthModule, LogLevel, StsConfigHttpLoader, StsConfigLoader } from 'angular-auth-oidc-client';
+import { map } from 'rxjs/operators';
 
+export const httpLoaderFactory = (httpClient: HttpClient) => {
+  const config$ = httpClient.get<any>(`assets/oidc.json`).pipe(
+    map((customConfig: any) => {
+      return {
+        triggerAuthorizationResultEvent: true,
+        postLoginRoute: customConfig.postLoginRoute,
+        logLevel: LogLevel.Debug,
+        historyCleanupOff: true,
+        eagerLoadAuthWellKnownEndpoints: false,
+        authority: customConfig.authority,
```

Comment:
We can just use the spread operator to apply any settings in the config onto this configuration instead of explicitly setting each field.

Reply:
Hi Phil!, it is very good idea to use spread operator. I prefer to leave right now and will revise later on, when we will figure out what settings should be hard-coded, and which should be saved in oidc.json file, if you don't mind! :)